substrate/frame/system/src/extensions/check_nonce.rs

```diff
@@ -152,22 +163,43 @@ mod tests {
 			let info = DispatchInfo::default();
 			let len = 0_usize;
 			// stale
-			assert_noop!(
-				CheckNonce::<Test>(0u64.into()).validate(&1, CALL, &info, len),
-				InvalidTransaction::Stale
+			assert_eq!(
```

Comment:
why not keep using `assert_noop`?

Reply:
Missing `PartialEq` impl.

Reply:
There is still `assert_storage_noop` or the `StorageNoopGuard` to ensure that no storage was modified.

Reply:
I used those, thanks :+1:  !